pkg/virtctl/create/instancetype/instancetype.go

```diff
@@ -0,0 +1,262 @@
+	for _, param := range c.gpus {
+		gpu := GPU{}
+
+		err := params.Map(GPUFlag, param, &gpu)
+		if err != nil {
+			return err
+		}
+
+		if gpu.Name == "" {
+			return params.FlagErr(GPUFlag, NameErr)
+		}
+
+		if gpu.DeviceName == "" {
+			return params.FlagErr(GPUFlag, DeviceNameErr)
+		}
+
+		instancetypeSpec.GPUs = append(instancetypeSpec.GPUs, v1.GPU{Name: gpu.Name, DeviceName: gpu.DeviceName})
+	}
+
+	return nil
+}
+
+func withHostDevices(c *createInstancetype, instancetypeSpec *instancetypev1alpha2.VirtualMachineInstancetypeSpec) error {
+	for _, param := range c.hostDevices {
+		hostDevice := HostDevice{}
+
+		err := params.Map(HostDeviceFlag, param, &hostDevice)
+		if err != nil {
+			return err
+		}
+
+		if hostDevice.Name == "" {
+			return params.FlagErr(HostDeviceFlag, NameErr)
+		}
+
+		if hostDevice.DeviceName == "" {
+			return params.FlagErr(HostDeviceFlag, DeviceNameErr)
+		}
+
+		instancetypeSpec.HostDevices = append(instancetypeSpec.HostDevices, v1.HostDevice{Name: hostDevice.Name, DeviceName: hostDevice.DeviceName})
+	}
+
+	return nil
+}
+
+func withIOThreadsPolicy(c *createInstancetype, instancetypeSpec *instancetypev1alpha2.VirtualMachineInstancetypeSpec) error {
+	var policy v1.IOThreadsPolicy
+
+	switch c.ioThreadsPolicy {
+	case string(v1.IOThreadsPolicyAuto):
+		policy = v1.IOThreadsPolicyAuto
+	case string(v1.IOThreadsPolicyShared):
+		policy = v1.IOThreadsPolicyShared
+	default:
+		return params.FlagErr(IOThreadsPolicyFlag, IOThreadErr)
+	}
+
+	instancetypeSpec.IOThreadsPolicy = &policy
+	return nil
+}
+
+func defaultCreateInstancetype() createInstancetype {
+	return createInstancetype{
+		name: "instancetype-" + rand.String(5),
+	}
+}
+
+func (c *createInstancetype) usage() string {
+	return `  # Create a manifest for a Instancetype with a random name and default values:
+  {{ProgramName}} create instancetype
+  
+  # Create a manifest for a Instancetype with a specified name and cpu:
+  {{ProgramName}} create instancetype --name my-instancetype --cpu 2
+
+  # Create a manifest for a Instancetype with a specified memory and gpu:
+  {{ProgramName}} create instancetype --memory 256Mi --gpu name:gpu1,devicename:nvidia
+  
+  # Create a manifest for a ClusterInstancetype with a specified name and cpu:
+  {{ProgramName}} create instancetype --cluster --name my-instancetype --cpu 2`
+}
+
+func newInstancetype(c *createInstancetype) *instancetypev1alpha2.VirtualMachineInstancetype {
+	return &instancetypev1alpha2.VirtualMachineInstancetype{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "VirtualMachineInstancetype",
+			APIVersion: instancetypev1alpha2.SchemeGroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: c.name,
+		},
+		Spec: instancetypev1alpha2.VirtualMachineInstancetypeSpec{
+			CPU: instancetypev1alpha2.CPUInstancetype{
+				Guest: uint32(c.cpu),
+			},
+			Memory: instancetypev1alpha2.MemoryInstancetype{
+				Guest: resource.MustParse(c.memory),
```

Comment:
With user input it is not advisable to panic if the value cannot be parsed. Better check and return an error if memory is invalid.

Reply:
Agreed, could you just move the call to `ParseQuantity` into this method and use the returned `Quantity`?